src/components/TracePage/TraceTimelineViewer/ListView/Positions.js

```diff
@@ -0,0 +1,210 @@
+// @flow
+
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+/**
+ * Keeps track of the height and y-position for anything sequenctial where
+ * y-positions follow one-after-another and can be derived from the height of
+ * the prior entries. The height is known from an accessor function parameter
+ * to the methods that require new knowledge the heights.
+ *
+ * @export
+ * @class Positions
+ */
+export default class Positions {
+  /**
+   * Indicates how far past the explicitly required height or y-values should
+   * checked.
+   * @type {number}
```

Comment:
Just an FYI, I think flow types will give you this for free so you don't need to write it.

Reply:
Thanks. Turns out jsDoc is dead to me, long live [documentation.js](https://github.com/documentationjs/documentation/blob/master/docs/GETTING_STARTED.md#flow-type-annotations). It's got a ton of problems, but it looks like you get a lot for free :+1: